src/kOS/Module/kOSProcessor.cs

```diff
@@ -338,7 +338,18 @@ public void InitObjects()
                     var bootVolumeFile = archive.Open(bootFile);
                     if (bootVolumeFile != null)
                     {
-                        HardDisk.Save(bootFile, bootVolumeFile.ReadAll());
+                        FileContent content = bootVolumeFile.ReadAll();
+                        if (HardDisk.IsRoomFor(bootFile, content))
+                        {
+                            HardDisk.Save(bootFile, bootVolumeFile.ReadAll());
```

Comment:
good point, I added that variable to improve performance and then forgot to remove the original call.  I'll update that.